src/antsibull_changelog/toml.py

```diff
@@ -0,0 +1,55 @@
+# Author: Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2023, Ansible Project
+
+"""
+TOML handling.
+"""
+
+from __future__ import annotations
+
+from typing import Any
+
+try:
+    import tomllib
+    HAS_TOMLLIB = True
+except ImportError:
+    HAS_TOMLLIB = False
+
+try:
+    import toml
+    HAS_TOML = True
+except ImportError:
+    HAS_TOML = False
+
+try:
+    import tomli
+    HAS_TOMLI = True
+except ImportError:
+    HAS_TOMLI = False
+
+
+def has_toml_loader_available() -> bool:
+    """
+    Load and parse YAML file ``path``.
+    """
+    return HAS_TOMLLIB or HAS_TOML or HAS_TOMLI
+
+
+def load_toml(path: str) -> Any:
+    """
+    Load and parse YAML file ``path``.
+    """
+    if HAS_TOMLLIB:
+        with open(path, 'rb') as f:
+            return tomllib.load(f)
+    if HAS_TOML:
+        with open(path, 'r', encoding='utf-8') as f:
+            return toml.loads(f.read())
+    if HAS_TOMLI:
+        with open(path, 'r', encoding='utf-8') as f:
+            return tomli.loads(f.read())
```

Comment:
```suggestion
            return toml.load(f)
```

Reply:
This branch is for `tomli`, not for `toml`.

Reply:
Actually, both tomli and toml should use `.load(f)`.

Reply:
At least one of them did fail in the past with `.load(f)`.

Reply:
`tomli.load()` probably failed because you didn't update https://github.com/ansible-community/antsibull-changelog/blob/main/stubs/tomli.pyi.

Reply:
That would have been a typing error, but not a runtime error.

It looks like it was `toml.load()` that failed, I had to do this in the original PR: https://github.com/ansible-community/antsibull-changelog/pull/80/files#diff-524d69ea9e2f19154eab8133adc939ab41048c6144e5b1ef72a653722d0cecb5R434

Also tomli did not accept binary IO not *that* long ago (https://github.com/hukkin/tomli/commit/b9cbbe27b574c1e0c60112b9b763fef38b46346c), and nowadays no longer supports text IO.

Reply:
Fair enough. It makes sense to use the more widely compatible `toml.loads()` and `tomli.loads()`.